src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ProducerConsumerStream.cs

```diff
@@ -20,7 +20,6 @@ internal class ProducerConsumerStream : Stream
         public ProducerConsumerStream()
         {
             _buffer = new TaskCompletionSource<WriteBufferWrapper>();
-            _dataAvail = new TaskCompletionSource<int>();
```

Comment:
This would require Dispose() being called before any calls to WriteAsync for there to be an issue. This could happen if a MessageEncoder throws an exception before it ever gets to call WriteAsync so this shouldn't be a problem with our own encoders, but might be for external encoders. Although I don't like the extra allocation, I prefer it over an added null check in Flush() so I'll put that line back.